src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/HuggingFaceEmbeddingClient.java

```diff
@@ -0,0 +1,54 @@
+package io.stargate.sgv2.jsonapi.service.embedding.operation;
+
+import io.quarkus.rest.client.reactive.QuarkusRestClientBuilder;
+import jakarta.ws.rs.HeaderParam;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import java.net.URI;
+import java.util.List;
+import org.eclipse.microprofile.rest.client.annotation.ClientHeaderParam;
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+
+public class HuggingFaceEmbeddingClient implements EmbeddingService {
+  private String apiKey;
+  private String modelName;
+
+  private String baseUrl;
+  private final HuggingFaceEmbeddingService embeddingService;
+
+  public HuggingFaceEmbeddingClient(String baseUrl, String apiKey, String modelName) {
+    this.apiKey = apiKey;
+    this.modelName = modelName;
+    this.baseUrl = baseUrl;
+    embeddingService =
+        QuarkusRestClientBuilder.newBuilder()
+            .baseUri(URI.create(baseUrl))
+            .build(HuggingFaceEmbeddingService.class);
+  }
+
+  public HuggingFaceEmbeddingClient(String apiKey, String modelName) {
+    this("https://api-inference.huggingface.co", apiKey, modelName);
```

Comment:
Should maybe define in one of constants classes? (if there's natural place for it) -- I think config class duplicates this as default URL? (so maybe define as constant there actually).

Reply:
Removed these constructor as they are defined as part of property.